### DISCUSSION TITLE
How to create a child prefix with specific starting value and netmask in a custom script?

### DISCUSSION THREAD

**Discussion by paulcarre07** [netbox-community/netbox#18249] [Help Wanted!] [answered]
Hello everyone,
In a custom script, I would like to create a child prefix from a given parent prefix. The child prefix should have its last octet set to 232, and its CIDR should be /29.

Is there a built-in function in NetBox that allows creating a prefix starting at a specific value and choosing the netmask? Or do I need to calculate and create it manually?

Thank you for your help!

**Comment by candlerb** [comment, marked as answer]
You simply have to create (and save) a Prefix object with the desired parameters.

There is no explicit linking of "Parent" and "Child" prefixes; this is inferred from the prefix being contained within a larger prefix, at query time.

**Comment by paulcarre07** [comment]
Thank you for the clarification! I understand now that I just need to create and save a Prefix object with the desired parameters, and the parent-child relationship is determined automatically based on containment.

I appreciate your help!
